### ISSUE TITLE
[NFW] NRE in ITypeSymbolExtensions.IsEnumType

### ISSUE THREAD

**Issue by amcasey** [dotnet/roslyn#9534]
Seen 4 times in 14.0.25*

```
Microsoft.CodeAnalysis.Shared.Extensions.ITypeSymbolExtensions::Boolean IsEnumType(Microsoft.CodeAnalysis.ITypeSymbol)
Microsoft.CodeAnalysis.Shared.Utilities.AbstractSpeculationAnalyzer`8[TSyntaxNode,TExpressionSyntax,TTypeSyntax,TAttributeSyntax,TArgumentSyntax,TForEachStatementSyntax,TThrowStatementSyntax,TSemanticModel]::Boolean SymbolsAreCompatibleCore(Microsoft.CodeAnalysis.ISymbol, Microsoft.CodeAnalysis.ISymbol, Boolean, Boolean)
Microsoft.CodeAnalysis.Shared.Utilities.AbstractSpeculationAnalyzer`8[TSyntaxNode,TExpressionSyntax,TTypeSyntax,TAttributeSyntax,TArgumentSyntax,TForEachStatementSyntax,TThrowStatementSyntax,TSemanticModel]::Boolean SymbolInfosAreCompatible(Microsoft.CodeAnalysis.SymbolInfo, Microsoft.CodeAnalysis.SymbolInfo, Boolean, Boolean)
Microsoft.CodeAnalysis.Shared.Utilities.AbstractSpeculationAnalyzer`8[TSyntaxNode,TExpressionSyntax,TTypeSyntax,TAttributeSyntax,TArgumentSyntax,TForEachStatementSyntax,TThrowStatementSyntax,TSemanticModel]::Boolean SymbolInfosAreCompatible(Microsoft...
```

Format is `DeclaringType::ReturnType Signature`.

**Comment by Pilchie** [comment]
Fixed by #17069

**Comment by jcouv** [comment]
Can we close this issue? @Pilchie referenced a fix above.
